### PR TITLE
Cleanup CMakeLists.txt for ExampleDesign samples

### PR DESCRIPTION
**Pull request by alainlou**
## Description

Refactor CMakeLists.txt for example designs

## Type of change
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- Manual testing for emulator and report targets for Linux on DevCloud
- Manual testing for emulator and report targets for Windows on laptop
- Have not gone through hw flow for any one of them

- [X] Command Line
- [ ] oneapi-cli
- [ ] Visual Studio
- [ ] Eclipse IDE
- [X] VSCode

### CODE REVIEW

**Comment by alainlou** [comment]
@mdbtucker

File: DirectProgramming/DPC++FPGA/ReferenceDesigns/crr/src/CMakeLists.txt [hunk 1]
```diff
@@ -2,104 +2,97 @@ set(SOURCE_FILE main.cpp)
...
 else()
-    message(FATAL_ERROR "Unknown board!")
+    message(FATAL_ERROR "Unknown or unsupported board!")
 endif()
 
 # Flags
-set(EMULATOR_COMPILE_FLAGS "-fintelfpga -DFPGA_EMULATOR -DOUTER_UNROLL=${OUTER_UNROLL} -DINNER_UNROLL=${INNER_UNROLL} -DOUTER_UNROLL_POW2=${OUTER_UNROLL_POW2}")
+# This is a Windows-specific flag that enables exception handling in host code
+if(WIN32)
+    set(WIN_FLAG "/EHsc")
```
Comment (mdbtucker) [inline DirectProgramming/DPC++FPGA/ReferenceDesigns/crr/src/CMakeLists.txt:36]: Please use -EHsc not /EHsc

File: DirectProgramming/DPC++FPGA/ReferenceDesigns/db/src/CMakeLists.txt [hunk 2]
```diff
@@ -47,118 +44,80 @@ endif()
...
-# FPGA hardware
+# This is a Windows-specific flag that enables exception handling in host code
 if(WIN32)
-    add_custom_target(fpga COMMAND echo "An FPGA hardware target is not provided on Windows. See README for details.")
-else()
-    add_executable(${FPGA_TARGET} EXCLUDE_FROM_ALL ${HOST_SOURCE} ${DEVICE_SOURCE})
-    set_target_properties(${FPGA_TARGET} PROPERTIES COMPILE_FLAGS ${HARDWARE_COMPILE_FLAGS})
-    set_target_properties(${FPGA_TARGET} PROPERTIES LINK_FLAGS ${HARDWARE_LINK_FLAGS})
-    add_custom_target(fpga DEPENDS ${FPGA_TARGET})
+    set(WIN_FLAG "/EHsc")
```
Comment (mdbtucker) [inline DirectProgramming/DPC++FPGA/ReferenceDesigns/db/src/CMakeLists.txt:67]: Please use -EHsc not /EHsc

File: DirectProgramming/DPC++FPGA/ReferenceDesigns/gzip/src/CMakeLists.txt [hunk 3]
```diff
@@ -16,214 +16,126 @@ set(EMULATOR_TARGET_LL ${TARGET_NAME}_ll.fpga_emu)
...
-
-    # Low-Latency Variant
-    if (NOT ${IS_A10})
-      add_executable(${EMULATOR_TARGET_LL} ${DEVICE_SOURCE_FILE_LL} ${HOST_SOURCE_FILE_LL})
-      add_custom_target(fpga_emu_ll DEPENDS ${EMULATOR_TARGET_LL})
-      set_target_properties(${EMULATOR_TARGET_LL} PROPERTIES COMPILE_FLAGS ${EMULATOR_COMPILE_FLAGS})
-      set_target_properties(${EMULATOR_TARGET_LL} PROPERTIES LINK_FLAGS ${EMULATOR_LINK_FLAGS})
-    endif()
-
+    set(WIN_FLAG "/EHsc")
```
Comment (mdbtucker) [inline DirectProgramming/DPC++FPGA/ReferenceDesigns/gzip/src/CMakeLists.txt:49]: Please use -EHsc not /EHsc

File: DirectProgramming/DPC++FPGA/ReferenceDesigns/qrd/src/CMakeLists.txt [hunk 4]
```diff
@@ -5,128 +5,95 @@ set(TARGET_NAME qrd)
...
-                       DEPENDS ${HOST_SOURCE_FILE})
-
-    add_custom_command(OUTPUT ${DEVICE_IMAGE_FPGA_OBJ}
-                       COMMAND ${CMAKE_CXX_COMPILER} ${CMAKE_CXX_FLAGS_LIST} ${HARDWARE_LINK_FLAGS} -fsycl-link=image ${DEVICE_FPGA_OBJ} -o ${DEVICE_IMAGE_FPGA_OBJ}
-                       DEPENDS ${DEVICE_FPGA_OBJ})
-
-    add_custom_command(OUTPUT ${FPGA_TARGET}
-                       COMMAND ${CMAKE_CXX_COMPILER} ${CMAKE_CXX_FLAGS_LIST} ${FINAL_LINK_FLAGS} ${HOST_FPGA_OBJ} ${DEVICE_IMAGE_FPGA_OBJ} -o  ${CMAKE_BINARY_DIR}/${FPGA_TARGET}
-                       DEPENDS ${DEVICE_IMAGE_FPGA_OBJ} ${HOST_FPGA_OBJ})
+    set(WIN_FLAG "/EHsc")
```
Comment (mdbtucker) [inline DirectProgramming/DPC++FPGA/ReferenceDesigns/qrd/src/CMakeLists.txt:40]: Please use -EHsc not /EHsc

**Review by mdbtucker — DISMISSED** [review]
Approving to trigger CI pipeline

**Comment by mdbtucker** [comment]
Alain, since you're touching the mvdr cmake file anyway, do you mind changing /EHsc to -EHsc there as well?

**Comment by tyoungsc** [comment]
LGTM. Thanks Alain!

File: DirectProgramming/DPC++FPGA/ReferenceDesigns/crr/src/CMakeLists.txt (hunk 1 above)
Comment (alainlou) [reply DirectProgramming/DPC++FPGA/ReferenceDesigns/crr/src/CMakeLists.txt]: sounds good

File: DirectProgramming/DPC++FPGA/ReferenceDesigns/db/src/CMakeLists.txt (hunk 2 above)
Comment (alainlou) [reply DirectProgramming/DPC++FPGA/ReferenceDesigns/db/src/CMakeLists.txt]: sounds good

File: DirectProgramming/DPC++FPGA/ReferenceDesigns/gzip/src/CMakeLists.txt (hunk 3 above)
Comment (alainlou) [reply DirectProgramming/DPC++FPGA/ReferenceDesigns/gzip/src/CMakeLists.txt]: sounds good

File: DirectProgramming/DPC++FPGA/ReferenceDesigns/qrd/src/CMakeLists.txt (hunk 4 above)
Comment (alainlou) [reply DirectProgramming/DPC++FPGA/ReferenceDesigns/qrd/src/CMakeLists.txt]: sounds good

**Comment by mdbtucker** [comment]
CI failure of mvdr_beamforming is expected, as the design has been updated to use 2021.3 syntax and the CI runs with 2021.2.

**Comment by tiwaria1** [comment]
LGTM, thanks Alain!

File: DirectProgramming/DPC++FPGA/ReferenceDesigns/gzip/src/CMakeLists.txt [hunk 5]
```diff
@@ -16,214 +16,126 @@ set(EMULATOR_TARGET_LL ${TARGET_NAME}_ll.fpga_emu)
...
     set(NUM_REORDER_LL "")
+    set(SUPPORTS_LL_VARIANT 1)
 else()
-    message(FATAL_ERROR "Unknown board!")
+    message(FATAL_ERROR "Unknown or unsupported board!")
 endif()
 
-
-# Compiler flags
-set(EMULATOR_COMPILE_FLAGS "-g0 -fintelfpga  -DFPGA_EMULATOR -DNUM_ENGINES=${NUM_ENGINES}")
```
Comment (alainlou) [inline DirectProgramming/DPC++FPGA/ReferenceDesigns/gzip/src/CMakeLists.txt:68]: @anandhv this happens just on Windows
Do we want to change it?
```
C:\Users\alou\repos\oneAPI-samples\DirectProgramming\DPC++FPGA\ReferenceDesigns\gzip\build>nmake fpga_emu

Microsoft (R) Program Maintenance Utility Version 14.28.29335.0
Copyright (C) Microsoft Corporation.  All rights reserved.

Scanning dependencies of target gzip.fpga_emu
[ 16%] Building CXX object src/CMakeFiles/gzip.fpga_emu.dir/gzipkernel.cpp.obj
dpcpp: warning: unknown argument ignored in clang-cl: '-g0' [-Wunknown-argument]
[ 33%] Building CXX object src/CMakeFiles/gzip.fpga_emu.dir/gzip.cpp.obj
dpcpp: warning: unknown argument ignored in clang-cl: '-g0' [-Wunknown-argument]
[ 50%] Building CXX object src/CMakeFiles/gzip.fpga_emu.dir/crc32.cpp.obj
dpcpp: warning: unknown argument ignored in clang-cl: '-g0' [-Wunknown-argument]
[ 66%] Building CXX object src/CMakeFiles/gzip.fpga_emu.dir/WriteGzip.cpp.obj
dpcpp: warning: unknown argument ignored in clang-cl: '-g0' [-Wunknown-argument]
[ 83%] Building CXX object src/CMakeFiles/gzip.fpga_emu.dir/CompareGzip.cpp.obj
dpcpp: warning: unknown argument ignored in clang-cl: '-g0' [-Wunknown-argument]
[100%] Linking CXX executable ..\gzip.fpga_emu.exe
[100%] Built target gzip.fpga_emu
```

File: DirectProgramming/DPC++FPGA/ReferenceDesigns/gzip/src/CMakeLists.txt [hunk 6]
```diff
@@ -16,214 +16,126 @@ set(EMULATOR_TARGET_LL ${TARGET_NAME}_ll.fpga_emu)
...
-set(EMULATOR_LINK_FLAGS -fintelfpga)
-
-# specify -MMD -fsycl-link-targets=... instead of -fintelfpga to workaround known issue; lower report quality 
-set(HARDWARE_COMPILE_FLAGS -MMD -fintelfpga -c -DNUM_ENGINES=${NUM_ENGINES})
```
Comment (alainlou) [inline DirectProgramming/DPC++FPGA/ReferenceDesigns/gzip/src/CMakeLists.txt:72]: @anandhv this happens just on Windows
Do we want to change it?
```
C:\Users\alou\repos\oneAPI-samples\DirectProgramming\DPC++FPGA\ReferenceDesigns\gzip\build>nmake report

Microsoft (R) Program Maintenance Utility Version 14.28.29335.0
Copyright (C) Microsoft Corporation.  All rights reserved.

Scanning dependencies of target gzip_report.a
[ 50%] Building CXX object src/CMakeFiles/gzip_report.a.dir/gzipkernel.cpp.obj
dpcpp: warning: unknown argument ignored in clang-cl '-MMD'; did you mean '-QMMD'? [-Wunknown-argument]
[100%] Linking CXX executable ..\gzip_report.a.exe
```

**Review by DmitryDenisenko751 — APPROVED** [review]

**Comment by mdbtucker** [comment]
Closing this request, individual Reference Design owners will re-submit these changes after testing.

**Pull request closed without merging**